webapp/src/components/channel_selector.tsx

```diff
@@ -0,0 +1,49 @@
+import React, {useCallback} from 'react';
+import {useSelector} from 'react-redux';
+
+import AsyncSelect from 'react-select/async';
+
+import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+
+import {getStyleForReactSelect} from '@/utils/styles';
+import {autocompleteUserChannels} from '@/actions';
+
+type SelectOption = {
+    label: string;
+    value: string;
+}
+
+type Props = {
+    onChange: (selected: string) => void;
+    value: string[];
+};
+
+export default function ChannelSelector(props: Props) {
+    const theme = useSelector(getTheme);
+
+    const loadOptions = useCallback(async (input: string): Promise<SelectOption[]> => {
+        const matchedChannels = await autocompleteUserChannels(input);
+
+        return matchedChannels.map((c) => ({
+            label: c.display_name,
+            value: c.id,
+        }));
+    }, []);
+
+    const handleChange = (selected: SelectOption) => {
+        props.onChange(selected.value);
+    };
```

Comment:
What happens when the select is cleared?

Reply:
`selected.value = string[]`